tough/src/schema/mod.rs

```diff
@@ -79,16 +95,39 @@ pub struct Signature {
 
 // =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
 
+/// TUF 4.3: The root.json file is signed by the root role's keys. It indicates which keys are
+/// authorized for all top-level roles, including the root role itself. Revocation and replacement
+/// of top-level role keys, including for the root role, is done by changing the keys listed for the
+/// roles in this file.
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 #[serde(tag = "_type")]
 #[serde(rename = "root")]
 pub struct Root {
+    /// A string that contains the version number of the TUF specification. Its format follows the
+    /// Semantic Versioning 2.0.0 (semver) specification.
     pub spec_version: String,
+
+    /// A boolean indicating whether the repository supports consistent snapshots. When consistent
+    /// snapshots is `true`, targets and certain metadata filenames are prefixed with either a
+    /// a version number or digest.
     pub consistent_snapshot: bool,
+
+    /// An integer that is greater than 0. Clients MUST NOT replace a metadata file with a version
+    /// number less than the one currently trusted.
     pub version: NonZeroU64,
+
+    /// Determines when metadata should be considered expired and no longer trusted by clients.
     pub expires: DateTime<Utc>,
+
+    /// A list of key IDs. The key ID must be correct for the specified key. Clients must calculate
```

Comment:
I like having the TUF spec language in the docstrings.  One nit - where RFC 2119 key words like "MUST" are used and capitalized in the spec, I think we should retain the caps, since they have standard meanings.